pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -0,0 +1,329 @@
+
+
+```js
+function initCheckPassword(el) {
+
+};
+```
+
+
+We’ll then see if the element we pass as our argument can pass the defined checks.
+
+
+```js
+function initCheckPassword(el) {
+ const checkPassword = () => {
+    let failed = false;
+    let successTest = passwordChecks.items.filter((item) => {
+      let passed = item.checkRegEx(el.value);
+       // passed logic 
+    });
+};
+```
+
+
+[Mutations within the `amp-script` component ]({{g.doc('/content/amp-dev/documentation/components/amp-script.md', locale=doc.locale).url.path}}?format=websites#mutations-and-user-actions) are generally triggered by a user action. This allows for `amp-script` to register [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventListener). 
+
+Our function will listen for two events, [`keyup`](https://developer.mozilla.org/en-US/docs/Web/API/Document/keyup_event) for when a user types into the input box, and [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) in case our user pastes their password.
+
+
+```js
+function initCheckPassword(el) {
+ const checkPassword = () => {
+    let failed = false;
+    let successTest = passwordChecks.items.filter((item) => {
+      let passed = item.checkRegEx(el.value);
+       // passed logic 
+    });
+}; 
+// is called when user types in input
+el.addEventListener("keyup", checkPassword);
+// is called if user pastes into input 
+el.addEventListener("change", checkPassword);  
+```
+
+
+Inside the `checkPassword` function, we’ll call two more:
+
+
+
+*   If the password passes a defined check, it will call `checkMet`, turning the font green. 
+*   If the password failed a defined check, it will call `checkRemoved`, turning the font red.
+
+```js
+function initCheckPassword(el) {
+  // removes invalid class and replaced with valid if check is met
+  const checkMet = (listItem) => {
+    listItem.classList.remove("invalid");
+    listItem.classList.add("valid");
+    return true
+  };
+  // removes valid class and replaced with valid if check is deleted
+  const checkRemoved = (listItem) => {
+    listItem.classList.remove("valid");
+    listItem.classList.add("invalid");
+    return false
+  }
+  // function goes over each check to see if its been met
+  const checkPassword = () => {
+    let failed = false;
+    let successTest = passwordChecks.items.filter((item) => {
+      let passed = item.checkRegEx(el.value);
+      // calls checkMet or checkRemoved
+      return passed ? checkMet(item.text) : checkRemoved(item.text);
+    });
+ }
+    // is called when user types in input
+    el.addEventListener("keyup", checkPassword);
+    // is called if user pastes into input 
+    el.addEventListener("change", checkPassword);
+};
+```
+
+
+
+Both of our check functions add and remove classes from the password requirement list items within the `rules` div. The `checkFailed` class will make our text red and the `checkPassed` will change it to green. 
```

Comment:
`checkFailed` & `checkPassed` are a bit confusing as you're adding `valid` and `invalid` class names above.